std/src/main/scala/algebra/std/list.scala

```diff
@@ -11,15 +12,21 @@ trait ListInstances {
 }
 
 class ListOrder[A](implicit ev: Order[A]) extends Order[List[A]] {
-  def compare(x: List[A], y: List[A]): Int =
-    (x, y) match {
-      case (Nil, Nil) => 0
-      case (Nil, _) => -1
-      case (_, Nil) => 1
-      case (x :: xs, y :: ys) =>
-        val n = ev.compare(x, y)
-        if (n != 0) n else compare(xs, ys)
-    }
+  def compare(xs: List[A], ys: List[A]): Int = {
+    @tailrec def loop(xs: List[A], ys: List[A]): Int =
+      xs match {
+        case Nil =>
```

Comment:
Alright - results! I added another "Faster" case that captures the best of all worlds (ie. only destructuring if both sides are non-empty):

```
Benchmark                                              Mode  Samples          Score          Error  Units
a.b.TupleMatchBenchmark.emptyListLeft2Match           thrpt       10  288748291.733 ± 17924410.135  ops/s
a.b.TupleMatchBenchmark.emptyListLeft2MatchFast       thrpt       10  296541912.711 ± 24067230.778  ops/s
a.b.TupleMatchBenchmark.emptyListLeft2MatchFaster     thrpt       10  285169229.660 ± 39657197.426  ops/s
a.b.TupleMatchBenchmark.emptyListLeftTuple            thrpt       10  289354315.848 ± 19768765.323  ops/s
a.b.TupleMatchBenchmark.emptyListRight2Match          thrpt       10  207099008.084 ± 12826978.962  ops/s
a.b.TupleMatchBenchmark.emptyListRight2MatchFast      thrpt       10  208560165.672 ±  8428833.060  ops/s
a.b.TupleMatchBenchmark.emptyListRight2MatchFaster    thrpt       10  307506512.715 ± 22808402.454  ops/s
a.b.TupleMatchBenchmark.emptyListRightTuple           thrpt       10  301994972.708 ±  7037205.320  ops/s
a.b.TupleMatchBenchmark.emptyLists2Match              thrpt       10  401533656.442 ±  6343634.820  ops/s
a.b.TupleMatchBenchmark.emptyLists2MatchFast          thrpt       10  392072876.831 ± 40274503.858  ops/s
a.b.TupleMatchBenchmark.emptyLists2MatchFaster        thrpt       10  403947568.500 ±  4337703.883  ops/s
a.b.TupleMatchBenchmark.emptyListsTuple               thrpt       10  402480381.315 ±  6077971.326  ops/s
a.b.TupleMatchBenchmark.nonEmptyLists2Match           thrpt       10  213825113.072 ±  4839351.734  ops/s
a.b.TupleMatchBenchmark.nonEmptyLists2MatchFast       thrpt       10  215357079.103 ±  2290166.270  ops/s
a.b.TupleMatchBenchmark.nonEmptyLists2MatchFaster     thrpt       10  222770062.760 ±  7090811.748  ops/s
a.b.TupleMatchBenchmark.nonEmptyListsTuple            thrpt       10  225791825.823 ±  3292351.042  ops/s
```